home/service/tracing/init.sh

```diff
@@ -22,24 +22,43 @@ main()
     fi
 }
 
-enable()
+start()
 (
     cd "$DIR"
 
+    run docker-compose -p my127ws-tracing pull
     run docker-compose -p my127ws-tracing up -d
+
+    local TRAEFIK_CONFIG="${DIR}/../proxy/traefik/root/traefik.toml"
+    if grep -q '\[inactive.tracing\]' "${TRAEFIK_CONFIG}"; then
+      cp "${TRAEFIK_CONFIG}" "${TRAEFIK_CONFIG}.before-tracing-active"
+      sed 's/\[inactive.tracing\]/\[tracing\]/' "${TRAEFIK_CONFIG}.before-tracing-active" > "${TRAEFIK_CONFIG}"
+      rm "${TRAEFIK_CONFIG}.before-tracing-active"
+      passthru ws global service proxy restart
+    fi
 )
 
-disable()
+stop()
 (
+    local DO_PROXY_RESTART="${1:-yes}"
     cd "$DIR"
 
+    local TRAEFIK_CONFIG="${DIR}/../proxy/traefik/root/traefik.toml"
+    if [ "$DO_PROXY_RESTART" = "yes" ] && grep -q '\[tracing\]' "${TRAEFIK_CONFIG}"; then
+      cp "${TRAEFIK_CONFIG}" "${TRAEFIK_CONFIG}.before-tracing-inactive"
+      sed 's/\[tracing\]/\[inactive.tracing\]/' "${TRAEFIK_CONFIG}.before-tracing-inactive" > "${TRAEFIK_CONFIG}"
+      rm "${TRAEFIK_CONFIG}.before-tracing-inactive"
+      if [ "$DO_PROXY_RESTART" = "yes" ]; then
+        passthru ws global service proxy restart
+      fi
+    fi
     run docker-compose -p my127ws-tracing down -v --rmi local
 )
 
 restart()
 {
-    disable
-    enable
+    stop "no"
```

Comment:
Why don't you restart proxy after stopping tracing? wouldn't it continue using the old config?

Reply:
Restart for this is the jaeger container itself, rather than the traefik container. Don't want to touch the traefik config or container at all.

Without this "no", we would remove the jaeger tracing lines from the traefik config, restart traefik, stop and remove the jaeger container, build and start the jaeger container, re-add the jaeger tracing lines to the traefik container and restart traefik again.